### PR TITLE
Merge custom TOC code into upstream

### PR DESCRIPTION
**Pull request by sharkmoos**
As you suggested in [Cov-ComSecCov-ComSec.github.io#58](https://github.com/Cov-ComSec/Cov-ComSec.github.io/pull/58) . I have removed the code changing the functionality of the TOC font matter, so this is just the shortcode code for adding a toc. 

As mentioned in [Cov-ComSecCov-ComSec.github.io#58](https://github.com/Cov-ComSec/Cov-ComSec.github.io/pull/58), the text is still blue. I will make it grey when I feel like spending more time trying random things hoping SCSS starts making sense

### CODE REVIEW

**Comment by jackorcherton** [comment]
Thanks Sharkmoo's I'll take a look and will investigate the current behaviour

**Comment by jackorcherton** [comment]
@sharkmoos, what do you mean by shortcode?

As mentioned in https://github.com/Cov-ComSec/Cov-ComSec.github.io/pull/58, I think this theme already has the functionality you need (or please correct me if I'm wrong?)

**Comment by sharkmoos** [comment]
Sure, let me run you through the change I'm proposing here.

From the [Cov-ComSec/Cov-ComSec.github.io/pull/58](https://github.com/Cov-ComSec/Cov-ComSec.github.io/pull/58), I stated the following, TOC can now be disabled via a page's font matter with the following line toc: false

> TOC can now be disabled via a page's font matter with the following line `toc: false`

This change not been suggested for this theme, as we are still testing the impact o

**Comment by sharkmoos** [comment]
Sure, let me run you through the change I'm proposing here.

From the https://github.com/Cov-ComSec/Cov-ComSec.github.io/pull/58, 

> TOC can now be disabled via a page's font matter with the following line `toc: false`

This change as not been suggested in this PR, as we are still testing the impact of the change


The change proposal here is around shortcodes. For your question about "what do you mean by shortcodes", I am talking about the Hugo template rendering wrappers (explained [here](https://gohugo.io/content-management/shortcodes/)). The purpose of the TOC shortcode proposed is to provide the functionality to place a TOC anywhere on a post, rather than it being generated at the top of the page. The TOC can be inserted into a page by providing the following shortcode in the source code of the page at the location the TOC should be rendered

```go
{{< toc >}}
```

**Comment by jackorcherton** [comment]
Ah ok, thanks for the explanation, this seems useful

I'll take a proper look when I get the chance before merging.

**Comment by jackorcherton** [comment]
Can confirm that the positioning works successfully with my site, however it seems to break the active colours when you hover over a link.

**Comment by sharkmoos** [comment]
Oh interesting. I think I have active colour contrasts disabled on my browser. I'll take a look at that, there may be a reasonable fix

**Comment by sharkmoos** [comment]
Can you send the colour code of the active colour?

**Comment by sharkmoos** [comment]
I pulled to colour code from [7e27d8e726556cbe23f7216b0e98cc9f768cebbb](https://github.com/Cov-ComSec/Cov-ComSec.github.io/commits/7e27d8e726556cbe23f7216b0e98cc9f768cebbb). This looks correct to you now?

File: assets/css/_partial/_shortcodes/toc.scss [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+      padding: .4rem 1rem .4rem 1.8rem;
+    }
+    a {
+      color: #a9a9b3;
+
+      [theme=dark] & {
+        color: #a9a9b3;
+      }
+      &::before {
+        color: $single-link-hover-color;
```
Comment (jackorcherton) [inline assets/css/_partial/_shortcodes/toc.scss]:
```suggestion
        color: $single-link-color;

```

File: assets/css/_partial/_shortcodes/toc.scss [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+        [theme=dark] & {
+          color: $single-link-hover-color-dark;
```
Comment (jackorcherton) [inline assets/css/_partial/_shortcodes/toc.scss]:
```suggestion
          color: $single-link-color-dark;

```

**Review by jackorcherton — CHANGES_REQUESTED** [review]
@sharkmoos Believe these 2 changes fix the menu colours.


I'll let you check you're happy with this, and if you are, we'll merge :)

**Comment by sharkmoos** [comment]
What is the change here? The colour of the vertical bar to the left of the text? 

ComSec website vs yours, is it slightly darker or something?

**Comment by jackorcherton** [comment]
Yep should be slightly darker to match the original theme.

Also, are you running it locally, as I haven't pushed this to my live site

**Comment by sharkmoos** [comment]
Sweet. No, not running locally. But your site has the [jackorcherton](https://github.com/jackorcherton) /
[HackIt](https://github.com/jackorcherton/HackIt) theme, and the comsec one is running the newest [Cov-ComSec](https://github.com/Cov-ComSec) /
[HackIt](https://github.com/Cov-ComSec/Cov-ComSec.github.io)

**Comment by sharkmoos** [comment]
These looks great then, I'll approve those

**Comment by jackorcherton** [comment]
The ComSec site is running with the changes I just made?

**Comment by jackorcherton** [comment]
Thought it should match now...

**Comment by sharkmoos** [comment]
No - your website has the *original* hackit theme, which is what we're trying to emulate? The ComSec website should have the theme with the incorrect vertical bar

**Comment by jackorcherton** [comment]
Yes but hopefully after this push they should be the same 🤞

**Comment by sharkmoos** [comment]
Sweet. Your update is done, feel free to merge

**Comment by jackorcherton** [comment]
Awesome

**Pull request merged**